src/common/safe_io.c

```diff
@@ -12,7 +12,9 @@
  *
  */
 
+#if !defined(__FreeBSD__)
 #define _XOPEN_SOURCE 500
+#endif
```

Comment:
i don't think this is a FreeBSD specific problem. `snprintf()` is part of C99 and C++11. let's just rename `safe_io.c`to `safe_io.cc` to avoid this problem.